File: packages/backend/sdk/tests/integration.test.ts
import { describe, expect, it } from 'vitest';
import { FluxisClient } from '../src/client.js';
import { getTestCredentials, loadTestEnv } from './testenv.js';

loadTestEnv();

const credentials = getTestCredentials();

function createIntegrationClient(): FluxisClient {
  if (!credentials) {
    throw new Error('Integration credentials are required');
  }

  return new FluxisClient({
    apiKey: credentials.apiKey,
    apiSecret: credentials.apiSecret,
  });
}

if (!credentials) {
  describe.skip('staging integration', () => {});
} else {
  describe('staging integration', () => {
    const client = createIntegrationClient();

    it('gets organization', async () => {
      const org = await client.organization.get();
      expect(org.id).toBeTruthy();
      expect(org.name).toBeTruthy();
    });

    it('lists accounts', async () => {
      const accounts = await client.accounts.list();
      expect(Array.isArray(accounts)).toBe(true);
    });

    it('lists point of sale entries', async () => {
      const result = await client.pointOfSale.list({ page: 1, pageSize: 10 });
      expect(result.page).toBeGreaterThanOrEqual(1);
      expect(Array.isArray(result.data)).toBe(true);
    });

    it('lists transactions', async () => {
      const result = await client.transactions.list({ page: 1, pageSize: 10 });
      expect(result.page).toBeGreaterThanOrEqual(1);
      expect(Array.isArray(result.data)).toBe(true);
    });

    it('rejects invalid NASPIP token via API', async () => {
      await expect(
        client.naspip.read('v4.local.invalid-token-for-integration-test'),
      ).rejects.toMatchObject({ name: 'FluxisError' });
    });

    it('creates and deletes a test account', async () => {
      const name = `sdk-ts-test-${crypto.randomUUID().slice(0, 8)}`;
      const account = await client.accounts.create({ name });
      expect(account.name).toBe(name);
      expect(account.id).toBeTruthy();
      await client.accounts.delete(account.id);
    });

    it('validates NASPIP token format locally', () => {
      expect(client.naspip.isValidTokenFormat('v4.local.placeholder')).toBe(true);
      expect(client.naspip.isValidTokenFormat('invalid')).toBe(false);
    });
  });
}

File: packages/backend/sdk-go/fluxis/integration_test.go
package fluxis

import (
	"context"
	"errors"
	"testing"
	"time"

	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
)

func TestIntegration(t *testing.T) {
	_ = LoadTestEnv()
	apiKey, apiSecret, ok := TestCredentials()
	if !ok {
		t.Skip("FLUXIS_API_KEY and FLUXIS_API_SECRET required for integration tests")
	}

	client, err := NewClient(
		WithAPIKey(apiKey),
		WithAPISecret(apiSecret),
	)
	require.NoError(t, err)

	ctx := context.Background()

	t.Run("Organization.Get", func(t *testing.T) {
		org, err := client.Organization.Get(ctx)
		require.NoError(t, err)
		assert.NotEmpty(t, org.ID)
		assert.NotEmpty(t, org.Name)
	})

	t.Run("Accounts.List", func(t *testing.T) {
		accounts, err := client.Accounts.List(ctx)
		require.NoError(t, err)
		assert.NotNil(t, accounts)
	})

	t.Run("PointOfSale.List", func(t *testing.T) {
		resp, err := client.PointOfSale.List(ctx, &ListPointOfSaleOptions{Page: 1, PageSize: 10})
		require.NoError(t, err)
		assert.GreaterOrEqual(t, resp.Page, 1)
	})

	t.Run("Transactions.List", func(t *testing.T) {
		resp, err := client.Transactions.List(ctx, &ListTransactionsOptions{Page: 1, PageSize: 10})
		require.NoError(t, err)
		assert.GreaterOrEqual(t, resp.Page, 1)
	})

	t.Run("Naspip.IsValidTokenFormat", func(t *testing.T) {
		assert.True(t, IsValidTokenFormat("v4.local.placeholder"))
	})

	t.Run("Naspip.Read_InvalidToken", func(t *testing.T) {
		_, err := client.Naspip.Read(ctx, "v4.local.invalid-token-for-integration-test")
		require.Error(t, err)
		var apiErr *FluxisError
		assert.True(t, errors.As(err, &apiErr))
	})

	t.Run("Accounts.CreateAndDelete", func(t *testing.T) {
		name := "sdk-go-test-" + time.Now().Format("150405")
		account, err := client.Accounts.Create(ctx, &CreateAccountRequest{Name: name})
		require.NoError(t, err)
		require.NotEmpty(t, account.ID)
		assert.Equal(t, name, account.Name)
		require.NoError(t, client.Accounts.Delete(ctx, account.ID))
	})
}
